### ISSUE TITLE
Invalid type error for tags

### ISSUE THREAD

**Issue by kakulukia** [buefy/buefy#3887]
### Overview of the problem

**Buefy** version: 0.9.23
**Vuejs** version: 2.6.14
**OS/Browser**: OSX / Chrome 

### Description

I wanted to try different tag types based on conditions. In this case i have a time table with bookable slots, reserved one and one selected item.

I used the following code to get this result:
` b-tag(:type="{ 'is-primary': !isBooked(day, slot), 'is-light': !isSelected(slot, day) }") [[ slot.start]] - [[ slot.end ]]` 

![SCR-20230919-aej](https://github.com/buefy/buefy/assets/96352/09348731-bcac-4aca-be20-5db5cc52605d)

But, even tho we see it working it is producing the following error:

![SCR-20230919-ajl](https://github.com/buefy/buefy/assets/96352/6c93672a-e185-46a3-89f8-49520821e503)

### Steps to reproduce

1. User a tag item with an object for its type rather than strings

### Expected behavior

As objects already work, the above error should not be triggered.

### Actual behavior

I see errors popping up AND Buefy accepting the object and rendering the type correctly.

I tried to set this [CodePen template](https://codepen.io/service-paradis/pen/KKgJZOK) up, but it wasnt displaying Buefy CSS for me.
Anyway, this works for me as some test code:

`<b-field><b-tag :type="{'is-primary': true}">test</b-tag><b-field>`

And interestingly it is producing two error for this one tag item.

**Comment by wesdevpro** [comment]
@kakulukia can you please share a code pen. Thanks

**Comment by kakulukia** [comment]
Hi @wesdevpro ,

i had a second try to create a codepen, but have a look: its not executing as it should.
![SCR-20231012-i1p](https://github.com/buefy/buefy/assets/96352/f4c1dacd-f977-490b-8788-fa7b1a4ef997)

Could you please provide a working empty template?
Here is the test code anyway: https://codepen.io/kakulukia/pen/vYvPwBN

**Comment by wesdevpro** [comment]
> Hi @wesdevpro ,
> 
> i had a second try to create a codepen, but have a look: its not executing as it should. ![SCR-20231012-i1p](https://user-images.githubusercontent.com/96352/274576411-f4c1dacd-f977-490b-8788-fa7b1a4ef997.png)
> 
> Could you please provide a working empty template? Here is the test code anyway: https://codepen.io/kakulukia/pen/vYvPwBN

Sure! It depends on what framework you are using.
Interesting.

**Comment by kakulukia** [comment]
yes, buefy and vue2

**Comment by kikuomax** [comment]
@kakulukia You are right. `type` is passed to `class` attribute of `<span>`s in `BTag`:
https://github.com/buefy/buefy/blob/181e9876258aaf7ba40fbaa4b56a33eca1f8e527/src/components/tag/Tag.vue#L5
https://github.com/buefy/buefy/blob/181e9876258aaf7ba40fbaa4b56a33eca1f8e527/src/components/tag/Tag.vue#L41

It may be any object that `class` can accept. So I think we can safely declare `type` may be `String` or `Object`.
